Lib/glyphsLib/builder/glyph.py

```diff
@@ -114,7 +114,53 @@ def to_ufo_glyph(self, ufo_glyph, layer, glyph):  # noqa: C901
         ufo_glyph.lib[GLYPHLIB_PREFIX + "subCategory"] = subCategory
 
     # load width before background, which is loaded with lib data
-    width = layer.width
+
+    # The width may be taken from another master via the customParameters
+    # 'Link Metrics With Master' or 'Link Metrics With First Master'.
+    master = self.font.masters[layer.associatedMasterId or layer.layerId]
+    source_master_id = master.customParameters[
+        "Link Metrics With Master"
+    ]
+    if source_master_id is None:
+        user_first_master = master.customParameters[
+            "Link Metrics With First Master"
+        ]
+        if user_first_master == 1:
+            width = self.font.glyphs[glyph.name].layers[0].width
+            if layer.width != width:
+                logger.info(
+                    f"{layer.parent.name}: Applying width from first master: "
+                    f"{layer.width} -> {width}"
+                )
+        else:
+            width = layer.width
+    else:
+        source_layer = self.font.glyphs[glyph.name].layers[source_master_id]
+        if source_layer is None:
+            # Try by name instead of master ID
+            found = False
+            for master in self.font.masters:
+                if master.name == source_master_id:
+                    found = True
+                    source_layer = self.font.glyphs[glyph.name].layers[master.id]
+                    break
+        else:
+            found = True
+        if found:
+            width = source_layer.width
+        else:
+            logger.debug(
```

Comment:
Could you switch these log levels (debug here, info later in the same file) around? If we're referencing a master and we don't have it, that's a problem and I want to know about it; if we're applying the width from a linked master, that's expected behaviour and I don't need to know about it.